### PR TITLE
Add support for mozilla and add stopListening method

### PR DESCRIPTION
**Pull request by bring2dip**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -15,12 +15,12 @@ export default class SpeechToText {
   */
   constructor(onAnythingSaid, onFinalised, onFinishedListening, language = 'en-US') {
     // Check to see if this browser supports speech recognition
-    if (!('webkitSpeechRecognition' in window)) {
-      throw new Error("This browser doesn't support speech recognition. Try Google Chrome.");
+    if (!('webkitSpeechRecognition' in window) || !('SpeechRecognition') in window) {
```
Comment (magician11) [inline src/index.js]: This line should read

```
    if (
      !('webkitSpeechRecognition' in window) &&
      !('SpeechRecognition' in window)
    ) {
```

Notice the `&&`. We only want to throw an error if both properties don't exist on the windows object.

Also you want to negate `('SpeechRecognition' in window)`, not just `('SpeechRecognition')`.
`!('SpeechRecognition') in window` returns false
`!('SpeechRecognition' in window)` returns true

**Comment by magician11** [comment]
Thanks for the input!

It looks like [the latest version of Firefox browser](https://developer.mozilla.org/en-US/docs/Web/API/SpeechRecognition#Browser_compatibility) still doesn't support this. Also, even though it says it supports Firefox mobile, I can't get it working on that too. But probably because you need to make changes to [the manifest file](https://developer.mozilla.org/en-US/docs/Web/API/SpeechRecognition#Firefox_OS_permissions).

You can view a demo using this module [here](https://apps.golightlyplus.com/speech-to-text-demo/).

No harm in putting the addition into the code though. Some changes required though.. I've made a note in your code.

And stopListening is a good idea.

File: src/index.js (hunk 1 above)
Comment (bring2dip) [reply src/index.js]: I'll make the changes and update the pr.

**Comment by rashmimhatre100** [comment]
Awesome package, But need stopListening method so that one can use it on component unmount. I am using this package for one of my project but really waiting for stopListening.

File: src/index.js (hunk 1 above)
Comment (bring2dip) [reply src/index.js]: I have updated the changes. And I am really sorry for the delay.

**Pull request merged**